public_api/src/network_links/network_links.ts

```diff
@@ -0,0 +1,40 @@
+import { Block, Dimension, Vector3 } from "@minecraft/server";
+import { NetworkLinkNode } from "./network_link_node.js";
+import { makeError } from "../internal.js";
+import { NETWORK_LINK_BLOCK_TAG, NETWORK_LINK_ENTITY_ID } from "./ipc_events.js";
+
+/**
+ * Finds the `NetworkLinkNode` associated with this block, or creates it if it doesn't exist
+ * 
+ * @param block Expected to have the `fluffyalien_energisticscore:network_link` block tag
+ * @returns The `NetworkLinkNode` at the block location
+ * @throws When the `NetworkLinkNode` does not exist yet, and the block at this location does not have the correct tags
+ */
+export function getNetworkLink(block: Block): NetworkLinkNode {
```

Comment:
the functions in this file can be turned into static methods on `NetworkLinkNode`, then the `NetworkLinkNode` constructor can be made private.